src/esw/brushless.py

```diff
@@ -108,28 +109,43 @@ async def _send_command(self) -> None:
         command = self._command
         self.command_lock.release()
         try:
-            state = await asyncio.wait_for(
-                self.controller.set_position(
-                    position=command.position,
-                    velocity=command.velocity,
-                    velocity_limit=CommandData.VELOCITY_LIMIT,
-                    maximum_torque=command.torque,
-                    watchdog_timeout=MoteusBridge.ROVER_NODE_TO_MOTEUS_WATCHDOG_TIMEOUT_S,
-                    query=True,
-                ),
-                timeout=self.MOTEUS_RESPONSE_TIME_INDICATING_DISCONNECTED_S,
-            )
+            await self._send_command(command)
+        except asyncio.TimeoutError:
+            if self.moteus_state.state != MoteusState.DISCONNECTED_STATE:
+                rospy.logerr(f"CAN ID {self._can_id} disconnected when trying to send command")
+            self._change_state(MoteusState.DISCONNECTED_STATE)
+
+    async def _send_command(self, command: CommandData) -> None:
+        """
+        Calls controller.set_position (which controls the moteus over CAN) with the requested command.
+        If the message has not been received within a certain amount of time, disconnected state is entered.
+        Otherwise, error state is entered.
+
+        It is expected that this may throw an error if there is a timeout.
+        """
+        state = await asyncio.wait_for(
+            self.controller.set_position(
+                position=command.position,
+                velocity=command.velocity,
+                velocity_limit=CommandData.VELOCITY_LIMIT,
+                maximum_torque=command.torque,
+                watchdog_timeout=MoteusBridge.ROVER_NODE_TO_MOTEUS_WATCHDOG_TIMEOUT_S,
+                query=True,
+            ),
+            timeout=self.MOTEUS_RESPONSE_TIME_INDICATING_DISCONNECTED_S,
+        )
+
+        moteus_not_found = state is None or not hasattr(state, "values") or moteus.Register.FAULT not in state.values
+        if moteus_not_found:
```

Comment:
Might be more readable to check for error (like in line 155) before calling _check_has_error() and then change _check_has_error() to _handle_error() or something like that

Reply:
changed